lib/posix-socket: Change argument type for posix_socket_close

flpostolache opened the pull request:
This PR fixes a possible crash in the socket, accept and socketpair syscall implementation when there is no fd available. It changes the structure given as argument to the
posix_socket_close function so that when calling this function it won't result in a crash.

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>

<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [ ] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.


### Base target

 - Architecture(s): N/A
 - Platform(s): N/A
 - Application(s): N/A


### Additional configuration

<!--
Please specify any additional configuration which is needed for this feature to
work or any new configuration parameters which are introduced by this PR.  This
will help during the review process.  For example:

 - `CONFIG_LIBUKDEBUG=y`

-->

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

craciunoiuc commented on lib/posix-socket/socket.c:
```diff
@@ -80,10 +81,16 @@ UK_SYSCALL_R_DEFINE(int, socket, int, family, int, type, int, protocol)
 	return ret;
 EXIT_ERR_CLOSE:
-	posix_socket_close(sock);
+		dummy = (struct posix_socket_file){
```
```suggestion
	dummy = (struct posix_socket_file){
```

flpostolache replied:
Changed.

marcrittinghaus reviewed (changes requested):
Hi @maniatro111,

thanks for submitting the bug fix! 😃 It looks good from my perspective. 

However, please remove the changes to the indentation. It seems you have your editor configured to 4 spaces per tab, but Unikraft uses 8 spaces per tab. The indentation is thus right, and your commit breaks it 😏

![image](https://user-images.githubusercontent.com/8928900/201855741-5a5053a6-1b42-4ed7-a832-2c581f464731.png)

craciunoiuc reviewed (changes requested):
Hey

You have an extra Tab on one of the lines.

Also, please specify in the PR and/or the commit that you also fixed some formatting in the file (probably this was done automatically by your editor :sweat_smile: )

marcrittinghaus commented:
> Hey
> 
> You have an extra Tab on one of the lines.
> 
> Also, please specify in the PR and/or the commit that you also fixed some formatting in the file (probably this was done automatically by your editor 😅 )

Good morning @craciunoiuc,

seems we were both looking at it at the same time 😏 As I said, formatting is generally right. So please revert changes. Maybe @craciunoiuc you have your editor not configured right as well 😆

flpostolache committed:
lib/posix-socket: Change argument type for posix_socket_close

This PR fixes a possible crash in the socket, accept and
socketpair syscall implementation when there is no fd available.
It changes the structure given as argument to the
posix_socket_close function so that when calling this function
it won't result in a crash.

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>

flpostolache commented:
Done, @marcrittinghaus and @craciunoiuc. Please have another look. I've disabled the auto indent and reconfigured the editor. Should be ok now.

craciunoiuc reviewed (approved):
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

marcrittinghaus reviewed (approved):
Thanks!

Approved-by: Marc Rittinghaus <marc.rittinghaus@unikraft.io>
